sdk/src/asset_handlers/jpeg_io.rs

```diff
@@ -571,6 +581,257 @@ impl RemoteRefEmbed for JpegIO {
+            (0xecu8, "APP12"),
+            (0xedu8, "APP13"),
+            (0xeeu8, "APP14"),
+            (0xefu8, "APP15"),
+            (0xfeu8, "COM"),
+            (0xc4u8, "DHT"),
+            (0xdbu8, "DQT"),
+            (0xddu8, "DRI"),
+            (0xd9u8, "EOI"),
+            (0xd0u8, "RST0"),
+            (0xd1u8, "RST1"),
+            (0xd2u8, "RST2"),
+            (0xd3u8, "RST3"),
+            (0xd4u8, "RST4"),
+            (0xd5u8, "RST5"),
+            (0xd6u8, "RST6"),
+            (0xd7u8, "RST7"),
+            (0xc0u8, "SOF0"),
+            (0xc2u8, "SOF2"),
+            (0xd8u8, "SOI"),
+            (0xdau8, "SOS"),
+            (0xf0u8, "JPG0"),
+            (0xf1u8, "JPG1"),
+            (0xf2u8, "JPG2"),
+            (0xf3u8, "JPG3"),
+            (0xf4u8, "JPG4"),
+            (0xf5u8, "JPG5"),
+            (0xf6u8, "JPG6"),
+            (0xf7u8, "JPG7"),
+            (0xf8u8, "JPG8"),
+            (0xf9u8, "JPG9"),
+            (0xfau8, "JPG10"),
+            (0xfbu8, "JPG11"),
+            (0xfcu8, "JPG12"),
+            (0xfdu8, "JPG13"),
+        ]);
+
+        let mut cai_en: Vec<u8> = Vec::new();
+        let mut cai_seg_cnt: u32 = 0;
+
+        let mut box_maps = Vec::new();
+        let mut curr_offset = 2; // start after JPEG marker
+
+        // load the bytes
+        let mut buf: Vec<u8> = Vec::new();
+        input_stream.rewind()?;
+        input_stream.read_to_end(&mut buf).map_err(Error::IoError)?;
+
+        // add first Map object contain SOI
+        let soi_bm = BoxMap {
+            names: vec!["SOI".to_string()],
+            alg: None,
+            hash: ByteBuf::from(Vec::new()),
+            pad: ByteBuf::from(Vec::new()),
+            range_start: 0,
+            range_len: 2,
+        };
+        box_maps.push(soi_bm);
+
+        let mut c2pa_bm_index = 0;
+
+        let dimg = DynImage::from_bytes(buf.into())
+            .map_err(|e| Error::OtherError(Box::new(e)))?
+            .ok_or(Error::UnsupportedType)?;
+
+        match dimg {
+            DynImage::Jpeg(jpeg) => {
+                for seg in jpeg.segments() {
+                    match seg.marker() {
+                        markers::APP11 => {
+                            // JUMBF marker
+                            let raw_bytes = seg.contents();
+
+                            if raw_bytes.len() > 16 {
+                                // we need at least 16 bytes in each segment for CAI
+                                let mut raw_vec = raw_bytes.to_vec();
+                                let _ci = raw_vec.as_mut_slice()[0..2].to_vec();
+                                let en = raw_vec.as_mut_slice()[2..4].to_vec();
+
+                                let is_cai_continuation = vec_compare(&cai_en, &en);
+
+                                if cai_seg_cnt > 0 && is_cai_continuation {
+                                    cai_seg_cnt += 1;
+
+                                    let c2pa_bm = box_maps.get_mut(c2pa_bm_index).ok_or(
```

Comment:
Are we sure that our map is exhaustive? It would be a shame to fail here if there are some exceptions.

Reply:
Not sure what you mean.  We have not written our own JPEG parser so the list is what is presented for the library.  The library does have the quirk/bug of now including the SOI and EOI segments.